zebra/zebra_vxlan.c

```diff
@@ -4786,24 +4804,39 @@ static int zl3vni_remote_nh_add(zebra_l3vni_t *zl3vni, struct ipaddr *vtep_ip,
 				struct prefix *host_prefix)
 {
 	char buf[ETHER_ADDR_STRLEN];
-	char buf1[INET6_ADDRSTRLEN];
+	char buf1[ETHER_ADDR_STRLEN];
+	char buf2[INET6_ADDRSTRLEN];
+	char buf3[PREFIX_STRLEN];
 	zebra_neigh_t *nh = NULL;
 
+	/* Create the next hop entry, or update its mac, if necessary. */
 	nh = zl3vni_nh_lookup(zl3vni, vtep_ip);
 	if (!nh) {
 		nh = zl3vni_nh_add(zl3vni, vtep_ip, rmac);
 		if (!nh) {
-
 			zlog_debug(
-				"Failed to add NH as Neigh (IP %s MAC %s L3-VNI %u)",
-				ipaddr2str(vtep_ip, buf1, sizeof(buf1)),
+				"Failed to add NH %s as Neigh (RMAC %s L3-VNI %u prefix %s)",
+				ipaddr2str(vtep_ip, buf1, sizeof(buf2)),
 				prefix_mac2str(rmac, buf, sizeof(buf)),
-				zl3vni->vni);
+				zl3vni->vni,
+				prefix2str(host_prefix, buf2, sizeof(buf2)));
 			return -1;
 		}
 
 		/* install the nh neigh in kernel */
 		zl3vni_nh_install(zl3vni, nh);
+	} else if (memcmp(&nh->emac, rmac, ETH_ALEN) != 0) {
+		if (IS_ZEBRA_DEBUG_VXLAN)
+			zlog_debug("L3VNI %u RMAC change(%s --> %s) for nexthop %s, prefix %s",
+				   zl3vni->vni,
+				   prefix_mac2str(&nh->emac, buf, sizeof(buf)),
+				   prefix_mac2str(rmac, buf1, sizeof(buf1)),
+				   ipaddr2str(vtep_ip, buf2, sizeof(buf2)),
+				   prefix2str(host_prefix, buf3, sizeof(buf3)));
+
+		memcpy(&nh->emac, rmac, ETH_ALEN);
```

Comment:
Same as above comment. We are updating the nh->emac here locally with new rmac. But where/when does the nh entry in hash (zl3vni->nh_table) gets updated with new mac address ? Even though we install, the nh entry in the hash table still has old mac value. Please check.

Reply:
The new change updates existing nh entry from hash with new rmac value. zl3vni->nh_table (hash) lookup retruns nh pointer which means existing nh, replace with new rmac value and install the new rmac into fdb table. 
With this change, performed extensive symmetric routing test where changing the rmac and/or nexthop value for evpn routes which are imported into vrf table. The nexthop and rmac fields are updated correctly into zebra's nh and rmac hash table.